src/Symfony/Component/Lock/LockFactory.php

```diff
@@ -11,12 +11,42 @@
 
 namespace Symfony\Component\Lock;
 
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+
 /**
  * Factory provides method to create locks.
  *
  * @author Jérémy Derussé <jeremy@derusse.com>
  * @author Hamza Amrouche <hamza.simperfit@gmail.com>
  */
-class LockFactory extends Factory
+class LockFactory
```

Comment:
Should it implement the `LoggerAwareInterface`?

Reply:
yes! done